airflow/jobs/triggerer_job_runner.py

```diff
@@ -676,11 +676,14 @@ def update_triggers(self, requested_trigger_ids: set[int]):
             try:
                 new_trigger_orm = new_triggers[new_id]
                 trigger_class = self.get_trigger_by_classpath(new_trigger_orm.classpath)
-            except BaseException as e:
-                # Either the trigger code or the path to it is bad. Fail the trigger.
-                self.failed_triggers.append((new_id, e))
+                new_trigger_instance = trigger_class(**new_trigger_orm.kwargs)
+            except (BaseException, TypeError) as err:
```

Comment:
```suggestion
            except BaseException as err:
```

Doesn’t TypeError inherit from BaseException?

```pycon
>>> issubclass(TypeError, BaseException)
True
```

Reply:
Yes, if that's the case, does it mean we catch an exception too broad for the following section? Or is it something expected?

```python
                new_trigger_orm = new_triggers[new_id]
                trigger_class = self.get_trigger_by_classpath(new_trigger_orm.classpath)
```

Reply:
No, the `except` block is fine, just you don’t need to mention `TypeError` at all here; the suggested change would just work.

Reply:
Just updated it as suggested. Thanks!